### ISSUE TITLE
ERROR: The remote host closed the connection. The error code is 0x800704CD. (System.Web.HttpException)

### ISSUE THREAD

**Issue by akuryan** [SitecoreUnicorn/Unicorn#387]
**Do you want to request a *feature* or report a *bug*?**
It is a bug
**What is the current behavior?**
When I try to sync Unicorn with some long running operation (I observed a project being synced for 500000 ms) via Unicorn.psm1 script - it generates an error 
```
5800 12:33:30 ERROR ERROR: The remote host closed the connection. The error code is 0x800704CD. (System.Web.HttpException)
at System.Web.Hosting.IIS7WorkerRequest.RaiseCommunicationError(Int32 result, Boolean throwOnDisconnect)
   at System.Web.Hosting.IIS7WorkerRequest.ExplicitFlush()
   at System.Web.HttpResponse.Flush(Boolean finalFlush, Boolean async)
   at System.Web.HttpWriter.Write(String s)
   at System.Web.HttpResponseWrapper.Write(String s)
   at Kamsar.WebConsole.TextWebConsole.Write(String statusMessage, MessageType type, Object[] formatParameters)
   at Unicorn.Logging.PubSubLogger.Flush()
   at Unicorn.Pipelines.UnicornSyncComplete.SyncCompleteLogger.Process(UnicornSyncCompletePipelineArgs args)
   at (Object , Object )
   at Sitecore.Pipelines.CorePipeline.Run(PipelineArgs args)
   at Sitecore.Pipelines.DefaultCorePipelineManager.Run(String pipelineName, PipelineArgs args, String pipelineDomain, Boolean failIfNotExists)
   at Sitecore.Pipelines.DefaultCorePipelineManager.Run(String pipelineName, PipelineArgs args, String pipelineDomain)
   at Unicorn.SerializationHelper.RunUnicornSyncComplete(IConfiguration configuration, DateTime syncStartTimestamp)
   at Unicorn.SerializationHelper.SyncTree(IConfiguration configuration, Action`1 rootLoadedCallback, Boolean runSyncStartPipeline, IItemData partialSyncRoot)
   at Unicorn.SerializationHelper.SyncConfigurations(IConfiguration[] configurations, IProgressStatus progress, ILogger additionalLogger)
```
When running sync in browser - it is not reproducing

**If the current behavior is a bug, please provide the steps to reproduce.**

**What is the expected behavior?**
Is there a way to continue sync in background if Stream could not be read and get sync status via API call?

**Please mention your Sitecore version and Unicorn version.**
Unicorn version is 4.1.1
Sitecore versin 9.1.0
This issue is reproduced stably when running in geo-replicated master and core databases in failover group with Active-Passive

**Comment by akuryan** [comment]
I tried to hide this problem by increasing timeout in script:
```
	$responseStreamReader = new-object System.IO.StreamReader $responseStream
	$responseStreamReader.BaseStream.ReadTimeout = 10800000
	$responseStreamReader.BaseStream.WriteTimeout = 10800000
```
But, this does not helped.

One more thing - sometimes when this timeout is recorder - my web app is restarted as well :(

**Comment by akuryan** [comment]
One more thing - same error could be reproduced when running Sync-Unicorn directly on azure webapp SCM (so, it is not related to internet connection quality or some other drops)

**Comment by akuryan** [comment]
Also, I see these kind of errors thrown, which terminates my ASP.NET process completely:
```
ERROR Unhandled exception detected. The ASP.NET worker process will be terminated. 
Exception: System.NullReferenceException
Message: Object reference not set to an instance of an object.
Source: System.Web
   at System.Web.HttpWriter.BufferData(Byte[] data, Int32 offset, Int32 size, Boolean needToCopyData)
   at System.Web.HttpWriter.FlushCharBuffer(Boolean flushEncoder)
   at System.Web.HttpWriter.GetBufferedLength()
   at System.Web.HttpResponse.UpdateNativeResponse(Boolean sendHeaders)
   at System.Web.HttpResponse.Flush(Boolean finalFlush, Boolean async)
   at System.Web.HttpWriter.Write(String s)
   at System.Web.HttpResponseWrapper.Write(String s)
   at Kamsar.WebConsole.TextWebConsole.Write(String statusMessage, MessageType type, Object[] formatParameters)
   at Kamsar.WebConsole.WebConsole.<.ctor>b__8_0(Object arg)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.TimerQueueTimer.CallCallback()
   at System.Threading.TimerQueueTimer.Fire()
   at System.Threading.TimerQueue.FireNextTimers()
```

**Comment by cassidydotdk** [comment]
Fixed in #390
